File: Problems/338.counting-bits.py
# title: Counting Bits
# timestamp: 2025-05-20 11:19:34
# problemUrl: https://leetcode.com/problems/counting-bits/
# memory: 18.5 MB
# runtime: 3 ms

class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [0]

        for i in range(1, n + 1):
            res.append(res[i // 2] + (i & 1))
        
        return res

File: Problems/338.counting-bits.go
// title: Counting Bits
// timestamp: 2024-04-18 13:18:34
// problemUrl: https://leetcode.com/problems/counting-bits/
// memory: 4.7 MB
// runtime: 5 ms

func countBits(n int) []int {
    ans := make([]int, n + 1)
    ans[0] = 0

    for i := 1; i <= n; i++ {
        ans[i] = ans[i>>1] + (i & 1)
    }

    return ans
}